tests/cli-tester.rs

```diff
@@ -0,0 +1,75 @@
+use assert_cmd::Command;
+use std::fs::{remove_file, File};
+use std::io::{Read, Write};
+use std::path::PathBuf;
+use tempfile::NamedTempFile;
+
+// Exemplar JSON state that won't be affected by the formatter
+const STATE: &str = "\"test\"";
+
+fn create_state() -> PathBuf {
+    let mut json = NamedTempFile::new_in(".").unwrap();
+    write!(json, "{STATE}").unwrap();
+
+    json.keep().unwrap().1
+}
+
+fn read_state(path: &PathBuf) -> String {
+    let mut file = File::open(path).unwrap();
+    let mut contents = String::new();
+    file.read_to_string(&mut contents).unwrap();
+
+    contents
+}
+
+#[test]
+fn test_no_clobber() {
```

Comment:
Add one more test about being able to create a new output file, and I suppose it's good.

Reply:
Will do :+1:

Hopefully I won't generate too much noise trying to track down the CI problem...